src/analysis/bounded_variation.lean

```diff
@@ -206,6 +196,33 @@ begin
   simp [u, edist_comm],
 end
 
+lemma eq_zero_iff (f : α → E) {s : set α} :
+  evariation_on f s = 0 ↔ ∀ (x y ∈ s), edist (f x) (f y) = 0 :=
+begin
+  split,
+  { rintro h x xs y ys,
+    rw [←le_zero_iff, ←h],
+    exact edist_le f xs ys, },
+  { rintro h,
+    rw [←le_zero_iff],
+    dsimp [evariation_on],
```

Comment:
`dsimp` shouldn't be needed before `apply` ... (untested)
Also, there's [finset.sum_eq_zero](https://leanprover-community.github.io/mathlib_docs/algebra/big_operators/basic.html#finset.sum_eq_zero)
```suggestion
```